src/snapred/backend/recipe/algorithm/SmoothDataExcludingPeaksAlgo.py

```diff
@@ -121,7 +121,12 @@ def PyExec(self):
 
         for index in range(numSpec):
             if np.all(weightWorkspace.readY(index) == 1):
-                smoothing_results = np.zeros_like(weightWorkspace.readY(index))
+                self.mantidSnapper.CloneWorkspace(
+                    "Cloning workspace for no peaks...",
+                    InputWorkspace=self.inputWorkspaceName,
+                    outputWorkspace=self.outputWorkspaceName,
+                )
+                self.mantidSnapper.executeQueue()
```

Comment:
Did Malcolm say to leave the workspace smoothed and not show a line at zero?

``` python
for index in range(numSpec):
    if not np.all(weightWorkspace.readY(index) == 1):
        x = inputWorkspace.readX(index)
        etc.
```

What you're doing here will re-create the output workspace from the input workspace.  That will end up overwriting results from previous `index` values.

Reply:
Or 
``` python
for index in range(numSpec):
    if np.any(weightWorkspace.readY(index) != 1):
        x = inputWorkspace.readX(index)
        etc.
```
or
``` python
for index in range(numSpec):
    if np.all(weightWorkspace.readY(index) == 1):
        continue
    else:
        x = inputWorkspace.readX(index)
        etc.
```

Reply:
Actually, removing the `if` entirely will do the same thing.

The values `outputWorkspace.readY(index)` are already equal to `inputWorkspace.readY(index)` because the output workspace was cloned from the input in L100.  If every value of `weightWorkspace.readY(index)` is equal to `1`, then at L136 `y = y[weightY != 0]` is  setting `y` equal to itself.  You then make a spline with all of the y-values, then get fitted values at the nodes of the spline, which means it return the exact same values.

So there isn't a need to branch if all of the weights are `1`.

However, it will save unecessary overhead calculations, as if all weights are `1` then you don't need to make a spline just to get back the node values.